### PR TITLE
feat: Implementing Chat Search UIs - Part 1

### PR DESCRIPTION
**Pull request by ice-kreios**
### What does this PR do?
 This PR introduces a unified main search feature that supports both chat and feed functionalities, leveraging shared UIs and logic. The feed_search has been refactored to accommodate both chat-related searches (such as people, groups, and channels) and feed-related searches (including top, latest, photos, and videos).
### Changes brought by this PR:

- Created shared UIs and logic for both feed and chat search functionalities.

- Enhanced AdvancedSearchCategory enum with isFeed and isChat properties to indicate the type of search category.

###  Additional Informations
Simulator Screenshot - iPhone 15 Pro Max - 2024-10-23 at 01 37 40
Simulator Screenshot - iPhone 15 Pro Max - 2024-10-23 at 01 35 54
Simulator Screenshot - iPhone 15 Pro Max - 2024-10-23 at 01 35 47

### CODE REVIEW

File: lib/app/features/wallet/views/pages/wallet_page/components/tabs/tabs_header_hide_action.dart [hunk 1]
```diff
@@ -28,9 +28,9 @@ class WalletTabsHeaderHideAction extends ConsumerWidget {
 
     return TextButton(
       onPressed: () {
-        final userId = ref.read(currentIdentityKeyNameSelectorProvider) ?? '';
+        final pubKey = ref.read(currentIdentityKeyNameSelectorProvider) ?? '';
```
Comment (ice-tychon) [inline lib/app/features/wallet/views/pages/wallet_page/components/tabs/tabs_header_hide_action.dart:31]: This one is not a `pubKey`, it's a `identityKeyName`.

File: lib/app/features/wallet/views/pages/wallet_page/components/nfts/nfts_header_layout_action.dart [hunk 2]
```diff
@@ -27,9 +27,9 @@ class NftHeaderLayoutAction extends ConsumerWidget {
       children: [
         TextButton(
           onPressed: () {
-            final userId = ref.read(currentIdentityKeyNameSelectorProvider) ?? '';
+            final pubKey = ref.read(currentIdentityKeyNameSelectorProvider) ?? '';
```
Comment (ice-tychon) [inline lib/app/features/wallet/views/pages/wallet_page/components/nfts/nfts_header_layout_action.dart:30]: `identityKeyName`

File: lib/app/features/wallet/views/pages/wallet_page/components/nfts/nfts_header_layout_action.dart [hunk 3]
```diff
@@ -47,9 +47,9 @@ class NftHeaderLayoutAction extends ConsumerWidget {
         ),
         TextButton(
           onPressed: () {
-            final userId = ref.read(currentIdentityKeyNameSelectorProvider) ?? '';
+            final pubKey = ref.read(currentIdentityKeyNameSelectorProvider) ?? '';
```
Comment (ice-tychon) [inline lib/app/features/wallet/views/pages/wallet_page/components/nfts/nfts_header_layout_action.dart:50]: `identityKeyName`

File: lib/app/features/wallet/views/pages/wallet_page/components/balance/balance.dart [hunk 4]
```diff
@@ -52,9 +52,9 @@ class Balance extends ConsumerWidget {
                     ),
                   ),
                   onPressed: () {
-                    final userId = ref.read(currentIdentityKeyNameSelectorProvider) ?? '';
+                    final pubKey = ref.read(currentIdentityKeyNameSelectorProvider) ?? '';
```
Comment (ice-tychon) [inline lib/app/features/wallet/views/pages/wallet_page/components/balance/balance.dart:55]: `identityKeyName`

File: lib/app/features/wallet/views/pages/nfts_sorting_modal/components/sorting_button/sorting_button.dart [hunk 5]
```diff
@@ -27,9 +27,9 @@ class SortingButton extends ConsumerWidget {
           color: context.theme.appColors.primaryAccent,
         ),
         onPressed: () {
-          final userId = ref.read(currentIdentityKeyNameSelectorProvider) ?? '';
+          final pubKey = ref.read(currentIdentityKeyNameSelectorProvider) ?? '';
```
Comment (ice-tychon) [inline lib/app/features/wallet/views/pages/nfts_sorting_modal/components/sorting_button/sorting_button.dart:30]: `identityKeyName`

**Review by ice-alcides — APPROVED** [review]

**Pull request merged**